### DISCUSSION TITLE
Read Video into OpenCV or Python (Headless) Raspberry PI 5

### DISCUSSION THREAD

**Discussion by D1no** [mikebrady/shairport-sync#1779] [Q&A] [answered]
Hi, I stumbled over this projects — which is amazing. I’m dabbling around with a machine learning toy project from which I take the iPhones screen and process it (i.e. OpenCV, plain Python/Rust, PyTorch, not sure yet).

Can anybody point me into the right direction of how to *get the video buffer programmatically* from the airplay mirror of shairport-sync? I have to admit that I am inexperienced in AV piping on Linux… hence this learning project 👀

Ideally I would like to be able to run shairport-sync on a headless Raspberry PI or Jetson Nano to get the airplay screen mirror this way. _Until I found this project, I was going to use a HDMI capture usb card. But this would be more awesome._

Thank you very much and happy new years!

**Comment by mikebrady** [comment]
Many thanks for your post, but unfortunately it doesn’t look as if Shairport Sync is suitable. The problem is that Shairport Sync is audio only — it doesn’t do any video. 🙁

**Reply by D1no** [reply]
Ah, thank you. Apparently I totally missed that. Mh. Are there any libraries out there that can act as an airplay receiver for video / screen mirroring? Thank you

**Reply by mikebrady** [reply, marked as answer]
Not that I know of, I'm afraid.
